### PR TITLE
Unignore cg_gcc fmt

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
`rustc_codegen_gcc` uses `rustfmt` now so it can be unignored.

r? @Urgau

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Is `x.py fmt` ignoring local `rustfmt.toml` files by any chance? :-/

When running `cargo fmt` on `rustc_codegen_gcc`, the code is different...

**Comment by Urgau** [comment]
We are using a different config than the one in `rustc_codegen_gcc`: https://github.com/rust-lang/rust/blob/b286722878e18db29a7fbe672be7c4d3b02e8e4d/rustfmt.toml#L1-L4

maybe that's the difference.

Also `./x.py fmt` is using a nightly `rustfmt` version: https://github.com/rust-lang/rust/blob/b286722878e18db29a7fbe672be7c4d3b02e8e4d/src/stage0#L19-L20

**Comment by GuillaumeGomez** [comment]
We're using nightly fmt as well and the config looks compatible with what we have:

```toml
use_small_heuristics = "Max"
```

No clue what's going on...

**Comment by Urgau** [comment]
Did you tried adding `version = "Two"`? I think this option changes some default behavior in `rustfmt`, which might be responsible for difference in formatting.

**Comment by Urgau** [comment]
You can put `#[rustfmt::ignore]` for those reluctant places.

File: compiler/rustc_codegen_gcc/rustfmt.toml
Comment (bjorn3) [inline compiler/rustc_codegen_gcc/rustfmt.toml:1]: `rustfmt.toml` should probably be changed to:

```toml
version = "Two"
use_small_heuristics = "Max"
merge_derives = false
```

to match the root rustfmt.toml.

**Comment by GuillaumeGomez** [comment]
cc @antoyo

**Review by antoyo — APPROVED** [review]
Those changes look good to me.

**Review by Urgau — APPROVED** [review]

**Pull request merged**